lib/mapObjects/CQuest.h

```diff
@@ -15,13 +15,25 @@
 #include "../CCreatureSet.h"
 #include "../NetPacksBase.h"
 
+namespace std
+{
+	// Used in std::unordered_map
+	template<> struct hash<VCMI_LIB_WRAP_NAMESPACE(ArtifactID)>
+	{
+		size_t operator()(const VCMI_LIB_WRAP_NAMESPACE(ArtifactID) & aid) const
+		{
+			return hash<int>{}(aid.num);
+		}
+	};
+}
```

Comment:
```suggestion
// Used in std::unordered_map
template<> struct std::hash<VCMI_LIB_WRAP_NAMESPACE(ArtifactID)>
{
	std::size_t operator()(const VCMI_LIB_WRAP_NAMESPACE(ArtifactID) & aid) const
	{
		return std::hash<int>{}(aid.num);
	}
};
```

Reply:
I already tried this case. It causes build fail

Reply:
made hash as structure inside ArtifactID